spyder/plugins/plots.py

```diff
@@ -0,0 +1,155 @@
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Plots Plugin"""
+
+
+# Third party imports
+from qtpy.QtCore import Signal
+from qtpy.QtWidgets import QStackedWidget, QGridLayout
+
+# Local imports
+from spyder.config.base import _
+from spyder.api.plugins import SpyderPluginWidget
+from spyder.api.preferences import PluginConfigPage
+from spyder.utils import icon_manager as ima
+from spyder.widgets.figurebrowser import FigureBrowser
+
+
+class PlotsConfigPage(PluginConfigPage):
+
+    def setup_page(self):
+        pass
+
+
+class Plots(SpyderPluginWidget):
+    """Plots plugin."""
+
+    CONF_SECTION = 'plots'
+    CONFIGWIDGET_CLASS = PlotsConfigPage
+    DISABLE_ACTIONS_WHEN_HIDDEN = False
+    sig_option_changed = Signal(str, object)
+
+    def __init__(self, parent):
+        SpyderPluginWidget.__init__(self, parent)
+
+        # Widgets
+        self.stack = QStackedWidget(self)
+        self.shellwidgets = {}
+
+        # Layout
+        layout = QGridLayout(self)
+        layout.addWidget(self.stack)
+
+        # Initialize plugin
+        self.initialize_plugin()
+
+    def get_settings(self):
+        """Retrieve all Plots configuration settings."""
+        return {name: self.get_option(name) for name in
+                ['mute_inline_plotting',  'show_plot_outline']}
+
+    # ---- Stack accesors
+
+    def set_current_widget(self, fig_browser):
+        """
+        Set the currently visible fig_browser in the stack widget, refresh the
+        actions of the cog menu button and move it to the layout of the new
+        fig_browser.
+        """
+        self.stack.setCurrentWidget(fig_browser)
+        # We update the actions of the options button (cog menu) and
+        # we move it to the layout of the current widget.
+        self.refresh_actions()
+        fig_browser.setup_options_button()
+
+    def current_widget(self):
+        return self.stack.currentWidget()
+
+    def count(self):
+        return self.stack.count()
+
+    def remove_widget(self, fig_browser):
+        self.stack.removeWidget(fig_browser)
+
+    def add_widget(self, fig_browser):
+        self.stack.addWidget(fig_browser)
+
+    # ---- Public API
+
+    def add_shellwidget(self, shellwidget):
+        """
+        Register shell with figure explorer.
+
+        This function opens a new FigureBrowser for browsing the figures
+        in the shell.
+        """
+        shellwidget_id = id(shellwidget)
+        if shellwidget_id not in self.shellwidgets:
+            self.options_button.setVisible(True)
+            fig_browser = FigureBrowser(
+                self, options_button=self.options_button)
+            fig_browser.set_shellwidget(shellwidget)
+            fig_browser.setup(**self.get_settings())
+            fig_browser.sig_option_changed.connect(
+                self.sig_option_changed.emit)
+            fig_browser.thumnails_sb.redirect_stdio.connect(
```

Comment:
This seems to be a grammatical error: `thumnails` should be `thumbnails`, right?